infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java

```diff
@@ -345,7 +351,17 @@ public void onClose(KubernetesClientException cause) {
    *     </ul>
    *     otherwise, it must be explicitly closed
    */
-  public CompletableFuture<Void> waitRunningAsync(String name) {
+  public CompletableFuture<Void> waitRunningAsync(String name, @Nullable Tracer tracer) {
```

Comment:
Why don't we inject tracer on object construction? It would simplify the code

Reply:
KubernetesDeployments isn't injected

Reply:
Than we can pass it at the object construction time. Might still make code cleaner

Reply:
This is worth considering.

`KubernetesDeployments` is created by `KubernetesNamespace`, which is created by the factory... etc. We would have to pass the tracer down all the way from `KubernetesRuntimeContext`, which seems more complex to just use it in this one method. OTOH, it would allow adding tracing to other KubernetesDeployments methods down the line.